### PR TITLE
Enhance 32-bit and 64-bit Number Operations in Assets

### PR DESCRIPTION
**Pull request by LucioDonda**
|Related issue|
|---|
|#17731|

**Description:**

### Overview:
This PR addresses the differentiation between 32-bit and 64-bit numbers in the Wazuh Engine's parsing of JSON and YAML. The goal is to provide a seamless experience for users, ensuring numbers are handled consistently regardless of their size.

### Changes:
1. **Dynamic Number Parsing:** Enhanced the parsing mechanism to dynamically choose the storage type based on the number's size. Numbers stored as 32-bit are now always retrieved and converted to 64-bit when necessary.
2. **Helper Function Updates:** Modified helper functions to support both 32-bit and 64-bit numbers.
3. **Backward Compatibility:** Ensured that the changes are backward compatible, eliminating the need for converting existing assets.
4. **Normalization Updates:** Modified the normalization stage in assets to set all values to true, aligning with the provided YAML example.

### Fixes:
- Resolved the issue where numbers exceeding 32 bits in YAML assets were stored as doubles in the resulting JSON.
- Enhanced helper functions to work seamlessly with both 32-bit and 64-bit numbers.

### Testing:
- Manually tested the changes with various assets to ensure consistent behavior.
- Verified that the helper functions now support both 32-bit and 64-bit numbers.

### Additional Notes:
- This enhancement aligns with the expected behavior, providing users with a consistent experience when working with numbers in assets.

### CODE REVIEW

**Review by JavierBejMen — APPROVED** [review]
LGTM

File: src/engine/source/builder/helperParser.hpp [hunk 1]
```diff
@@ -128,29 +128,31 @@ inline std::tuple<std::string, json::Json> toBuilderInput(const ExpressionToken&
         throw std::runtime_error("Expression field is empty");
     }
 
+    bool isValueIntiger = expressionToken.value.isInt() || expressionToken.value.isInt64();
```
Comment (LucioDonda) [inline src/engine/source/builder/helperParser.hpp:131]: Rename to `isValueInteger`

File: src/engine/test/source/json/json_test.cpp [hunk 2]
```diff
@@ -1042,6 +1042,36 @@ TEST_F(JsonGettersTest, GetInt64)
...
+        ASSERT_EQ(inputJson.getInt("/a").value(), number);
+    }
+
+    ASSERT_TRUE(inputJson.getIntAsInt64("/a").has_value());
+    ASSERT_EQ(inputJson.getIntAsInt64("/a").value(), number);
+}
+
+INSTANTIATE_TEST_SUITE_P(Json,
+                         IntAsInt64,
+                         ::testing::Values(0, 1, 2, 3, static_cast<int64_t>(std::numeric_limits<int32_t>::max()) + 10));
```
Comment (soynof) [inline src/engine/test/source/json/json_test.cpp:1073]: Add test with std::numeric_limits<int32_t>::min() -10, for example.

**Review by soynof — CHANGES_REQUESTED** [review]

File: src/engine/source/builder/helperParser.hpp (hunk 1 above)
Comment (juliancnn) [reply src/engine/source/builder/helperParser.hpp:131]: done

**Review by NahuFigueroa97 — APPROVED** [review]
LGTM!

**Review by JavierBejMen — APPROVED** [review]
LGTM

**Review by soynof — APPROVED** [review]
LGTM!

**Review by JcabreraC — APPROVED** [review]
LGTM !

**Pull request merged**